frontend/src/app/organization/organization-editor/organization-editor.component.ts

```diff
@@ -0,0 +1,136 @@
+import { profileResolver } from 'src/app/profile/profile.resolver';
+import { PermissionService } from 'src/app/permission.service';
+import { Organization, OrganizationService } from '../organization.service';
+import { Profile } from 'src/app/profile/profile.service';
+
+@Component({
+  selector: 'app-organization-editor',
+  templateUrl: './organization-editor.component.html',
+  styleUrls: ['./organization-editor.component.css']
+})
+export class OrganizationEditorComponent {
+  public static Route: Route = {
+    path: ':slug/edit',
+    component: OrganizationEditorComponent,
+    title: 'Organization Editor',
+    resolve: { profile: profileResolver }
+  };
+
+  /** Store the organization and its observable.  */
+  public organization$: Observable<Organization> | null = null;
+  public organization: Organization;
+
+  /** Store the currently-logged-in user's profile.  */
+  public profile: Profile | null = null;
+
+  /** Stores whether the user has admin permission over the current organization. */
+  public adminPermission: boolean = false;
+
+  /** Store the organization id. */
+  organization_slug: string = 'new';
+
+  /** Organization Editor Form */
+  public organizationForm = this.formBuilder.group({
+    name: "",
+    slug: "",
+    logo: "",
+    short_description: "",
+    long_description: "",
+    email: "",
+    website: "",
+    instagram: "",
+    linked_in: "",
+    youtube: "",
+    heel_life: "",
+    public: false
+  });
+
+  constructor(
+    private route: ActivatedRoute,
+    private router: Router,
+    protected formBuilder: FormBuilder,
+    protected snackBar: MatSnackBar,
+    private organizationService: OrganizationService,
+    private permission: PermissionService) {
+    /** Add validators to the form */
+    const form = this.organizationForm;
+    form.get('name')?.addValidators(Validators.required);
+    form.get('logo')?.addValidators(Validators.required);
+    form.get('short_description')?.addValidators(Validators.required);
+
+    /** Get currently-logged-in user. */
+    const data = this.route.snapshot.data as { profile: Profile };
+    this.profile = data.profile;
+
+    /** Get id from the url */
+    let organization_slug = this.route.snapshot.params['slug'];
+    this.organization_slug = organization_slug;
+
+    /** Set permission value if profile exists */
+    if (this.organization_slug == 'new') {
+      this.permission.check('admin.view', 'admin/').subscribe((perm) => this.adminPermission = perm)
+    }
+
+    /** Initialize organization */
+    this.organization = {
+      id: null,
+      name: "",
+      slug: "",
+      logo: "",
+      short_description: "",
+      long_description: "",
+      email: "",
+      website: "",
+      instagram: "",
+      linked_in: "",
+      youtube: "",
+      heel_life: "",
+      public: false
+    };
+  }
+
+  /** Event handler to handle submitting the Update Organization Form.
+   * @returns {void}
+   */
+  onSubmit = () => {

```

Comment:
We may be able to consider adding the slug generation that @KrisJordan was talking about for new organizations. The process would be as follows:
- Make string lowercase
- Using regex, select all non-alphanumeric characters and convert them to a - 
- Ensure that the slug does not already exist, otherwise as a suffix number.
    - Ex) If `cads` exists, the new org's slug would be `cads-2`.

Also, this means that we can remove the slug field entirely from the form and make its value calculated.

Reply:
I think Kris did say that the slugs should still be modifiable, but maybe we could auto generate an option for them based on their organization name. However, maybe we could save this for another ticket. I think there are a few more specifications we need to add to create nicer slugs, such as determining abbreviations. The current regex would just make "carolina-analytics-and-data-science" the slug.

Reply:
Good point! We can hold off for this commit then!